test/TestUtils.ts

```diff
@@ -0,0 +1,394 @@
+  EnvelopingRequest,
+  getEnvelopingRequestDataV4Field,
+  HttpClient,
+  HttpWrapper,
+  HubInfo,
+  isDeployRequest,
+  RelayRequest,
+  relayRequestType,
+} from '@rsksmart/rif-relay-client';
+import { ethers as hardhat } from 'hardhat';
+import { _TypedDataEncoder } from 'ethers/lib/utils';
+
+const SERVER_WORK_DIR = '/tmp/enveloping/test/server';
+const ONE_FIELD_IN_BYTES = 32;
+
+type StartRelayParams = {
+  serverConfig: ServerConfigParams;
+  delay?: number;
+  stake?: BigNumberish;
+  relayOwner: string;
+};
+
+const provider = hardhat.provider;
+
+const startRelay = async (relayHub: RelayHub, options: StartRelayParams) => {
+  const { serverConfig, delay, stake, relayOwner } = options;
+
+  fs.rmSync(SERVER_WORK_DIR, { recursive: true, force: true });
+
+  const originalConfig = { ...config };
+
+  config.util.extendDeep(originalConfig, {
+    ...serverConfig,
+    contracts: {
+      relayHubAddress: relayHub.address,
+    },
+    app: {
+      workdir: SERVER_WORK_DIR,
+      devMode: true,
+      checkInterval: 10,
+      logLevel: 5,
+    },
+  });
+
+  const url = config.get<string>('app.url');
+
+  const runServerPath = path.resolve(
+    __dirname,
+    '../node_modules/@rsksmart/rif-relay-server/dist/commands/Start.js'
+  );
+
+  const proc: ChildProcessWithoutNullStreams & { alreadyStarted?: number } =
+    childProcess.spawn('node', [runServerPath]);
+
+  const { relayManagerAddress } = await verifyRelayServerStatus(url, 3);
+
+  console.log('Relay Server Manager Address', relayManagerAddress);
+
+  await provider.getSigner(relayOwner).sendTransaction({
+    to: relayManagerAddress,
+    value: utils.parseEther('2'),
+  });
+
+  await relayHub.stakeForAddress(relayManagerAddress, delay || 2000, {
+    from: relayOwner,
+    value: stake || utils.parseEther('1'),
+  });
+
+  const { ready, relayWorkerAddress } = await verifyRelayServerStatus(
+    url,
+    25,
+    500,
+    true
+  );
+
+  expect(ready, 'Timed out waiting for relay to get staked and registered').is
+    .ok;
+
+  return {
+    proc,
+    worker: relayWorkerAddress,
+    manager: relayManagerAddress,
+  };
+};
+
+const stopRelay = (proc: ChildProcessWithoutNullStreams): void => {
+  proc.kill();
+};
+
+const evmMine = async (): Promise<unknown> => {
+  return provider.send('evm_mine', []);
+};
+
+const evmMineMany = async (count: number): Promise<void> => {
+  for (let i = 0; i < count; i++) {
+    await evmMine();
+  }
+};
+
+const increaseBlockchainTime = async (time: number): Promise<void> => {
+  await provider.send('evm_increaseTime', [time]);
+  await evmMine();
+};
+
+const createSnapshot = async (): Promise<string> => {
+  return (await provider.send('evm_snapshot', [])) as string;
+};
+
+const revertSnapshot = async (snapshotId: string) => {
+  return (await provider.send('evm_revert', [snapshotId])) as boolean;
+};
+
+const verifyRelayServerStatus = async (
+  url: string,
+  attempts: number,
+  interval = 1000,
+  isReady = false
```

Comment:
Why do we need the `isReady` param?

Reply:
because, we need to know first if the server is receiving requests and after staking check if its ready for transactions. Therefor I need to ping the server without being ready to get the managerAddress and then I need to ping it again once its ready after doing the stake process.

Reply:
Now I see what you mean, but what about creating an additional function for that specific case?
```ts
const getServerStatus = (url: string, client = new HttpClient(new HttpWrapper(undefined, 'silent'))) => client.getChainInfo(url);

// TODO: this function could be renamed to something like waitForServerToBeReady
const verifyRelayServerStatus = async (
  url: string,
  attempts: number,
  interval = 1000
): Promise<HubInfo> => {
  let response: HubInfo | undefined;
  const httpClient = new HttpClient(new HttpWrapper(undefined, 'silent'));

  for (let i = 0; i < attempts; i++) {
    try {
      console.log('sleep before cont.');
      await sleep(interval);
      response = await getServerStatus(url, httpClient);

      if (!response.ready) {
        continue;
      }

      return response;
    } catch (e) {
      /* empty */
     // I would log something here (I'm not a fan of empty catch blocks)
    }
  }

  throw Error("can't ping server");
};
```
So, instead of using it with the false flag, you could simply call `getServerStatus`. WDYT

Reply:
I thought about it, but was trying to avoid create duplication, but we could do that. Ill update it and push it.

Reply:
updated